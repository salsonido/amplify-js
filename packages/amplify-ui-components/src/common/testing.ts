import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
