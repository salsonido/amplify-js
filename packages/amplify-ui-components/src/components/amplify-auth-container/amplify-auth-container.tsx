import { Component, h, Host } from '@stencil/core';

/**
 * @slot (default) - Content placed within the container
 */
@Component({
	tag: 'amplify-auth-container',
})
export class AmplifyAuthContainer {
	render() {
		return (
			<Host>
				<form autoComplete="on" hidden>
					<input name="username"></input>
					<input name="password" type="password"></input>
					<input type="submit"></input>
				</form>
				<slot></slot>
			</Host>
		);
	}
}
