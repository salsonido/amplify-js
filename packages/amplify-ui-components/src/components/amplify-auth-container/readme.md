# amplify-auth-container



<!-- Auto Generated Below -->


## Slots

| Slot          | Description                         |
| ------------- | ----------------------------------- |
| `"(default)"` | Content placed within the container |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
