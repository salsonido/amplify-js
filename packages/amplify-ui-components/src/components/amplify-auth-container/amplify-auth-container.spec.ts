import { newSpecPage } from '@stencil/core/testing';
import { AmplifyAuthContainer } from './amplify-auth-container';

describe('amplify-auth-container spec:', () => {
  describe('Render logic ->', () => {
    it('should render with an empty slot', async () => {
      const page = await newSpecPage({
        components: [AmplifyAuthContainer],
        html: `<amplify-auth-container></amplify-auth-container>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });
});
