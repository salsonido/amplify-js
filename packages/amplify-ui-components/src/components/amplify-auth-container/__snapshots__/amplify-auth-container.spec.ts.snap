// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-auth-container spec: Render logic -> should render with an empty slot 1`] = `
<amplify-auth-container>
  <!---->
  <form autocomplete="on" hidden="">
    <input name="username">
    <input name="password" type="password">
    <input type="submit">
  </form>
</amplify-auth-container>
`;
